src/AutoRest.CSharp/Mgmt/Models/MgmtRestOperation.cs

```diff
@@ -315,11 +382,13 @@ internal static ResourceMatchType GetMatchType(HttpMethod httpMethod, RequestPat
             //catch check name availability where the provider ending matches
             //this one catches a lot so we are narrowing it down to containing "name" dont know all the checknameavailability name types
             if (requestLastSegment.IsConstant &&
-                requestPath.ToString()!.StartsWith("/subscriptions/{{subscriptionId}}", StringComparison.Ordinal) &&
-                requestLastSegment.ToString()!.Contains("name", StringComparison.OrdinalIgnoreCase) &&
+                requestPath.ToString().StartsWith("/subscriptions/{{subscriptionId}}", StringComparison.Ordinal) &&
```

Comment:
The previous code never works, all the request path will return false on this condition.
The `ToString` method of a request path instance will only write one `{}` around the reference, therefore we can only have `subscriptions/{subscriptionId}` instead of **{{**subscription**}}**